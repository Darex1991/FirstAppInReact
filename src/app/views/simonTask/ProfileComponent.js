import React, {Component, PropTypes} from 'react';
import { Form, Text } from 'react-form'
import './defaultInput.sass'

export default class ProfileComponent extends Component {
  constructor(props){
    super(props);
  }

  render() {
    return (
      <div className="profile default-input">
        <h1>Profile</h1>
        <div className="row">
          <div className="col-xs-6 m--b">
            <label>FirstName</label>
            <Text field='firstName' />
          </div>
          <div className="col-xs-6 m--b">
            <label>LastName</label>
            <Text field='lastName' />
          </div>
          <div className="col-xs-12 m--b">
            <label>Email</label>
            <Text field='email' type="email"/>
          </div>
          <div className="col-xs-6 m--b">
            <Text field='password' type="password" placeholder="Password"/>
          </div>
          <div className="col-xs-6 m--b">
            <Text field='password2' type="password" placeholder="Password confirm"/>
          </div>
        </div>
      </div>
    )
  }
}
