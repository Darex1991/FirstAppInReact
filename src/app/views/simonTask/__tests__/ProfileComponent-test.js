import React            from 'react';
import ReactTestUtils   from 'react-addons-test-utils'

import chai             from 'chai';
import chaiJestSnapshot from "chai-jest-snapshot";

import ProfileComponent from '../ProfileComponent';

chai.use(chaiJestSnapshot);

test('Should render ProfileComponent correctly', () => {
    const props = {
        dataBase: {
            phones: [
                {id: 1, name: 'asd', key: 1, value: 'val 1', checked: true, highPriority: true},
                {id: 2, name: 'asdasdasd', key: 2, value: 'val 122222', checked: false, highPriority: false},
                {id: 3, name: 'asdasdasd', key: 3, value: 'val asdasd', checked: false, highPriority: true},
                {id: 4, name: '222', key: 4, value: 'val213123 1', checked: true, highPriority: false}]
        },
        updateDataBase: (newDataBase) => {
            this.dataBase = newDataBase
        }
    };

    const shallowRenderer = ReactTestUtils.createRenderer();
    const result = shallowRenderer.render(<ProfileComponent {...props} />);
    expect(result).toMatchSnapshot();
});
