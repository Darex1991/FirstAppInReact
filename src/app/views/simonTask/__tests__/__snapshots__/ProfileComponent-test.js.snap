exports[`test Should render ProfileComponent correctly 1`] = `
<div
  className="profile default-input">
  <h2>
     Profile 
  </h2>
  <div
    className="row">
    <div
      className="col-xs-6 m--b">
      <label>
         FirstName 
      </label>
      <FormInputText
        field="firstName" />
    </div>
    <div
      className="col-xs-6 m--b">
      <label>
         LastName 
      </label>
      <FormInputText
        field="lastName" />
    </div>
    <div
      className="col-xs-12 m--b">
      <label>
         Email 
      </label>
      <FormInputText
        field="email"
        type="email" />
    </div>
    <div
      className="col-xs-6 m--b">
      <FormInputText
        field="password"
        placeholder="Password"
        type="password" />
    </div>
    <div
      className="col-xs-6 m--b">
      <FormInputText
        field="password2"
        placeholder="Password confirm"
        type="password" />
    </div>
  </div>
</div>
`;
